<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="David N. Nicholson" />
  <meta name="author" content="Faisal Alquaddoomi" />
  <meta name="author" content="Vincent Rubinetti" />
  <meta name="author" content="Casey S. Greene" />
  <meta name="dcterms.date" content="2022-07-08" />
  <meta name="keywords" content="pubtator central, preprints, natural language-processing, machine learning, time series, change point detection" />
  <title>Changing word meanings in biomedical literature reveal pandemics and new technologies</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Changing word meanings in biomedical literature reveal pandemics and new technologies" />
  <meta name="citation_title" content="Changing word meanings in biomedical literature reveal pandemics and new technologies" />
  <meta property="og:title" content="Changing word meanings in biomedical literature reveal pandemics and new technologies" />
  <meta property="twitter:title" content="Changing word meanings in biomedical literature reveal pandemics and new technologies" />
  <meta name="dc.date" content="2022-07-08" />
  <meta name="citation_publication_date" content="2022-07-08" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="David N. Nicholson" />
  <meta name="citation_author_institution" content="Genomics and Computational Biology Program, University of Pennsylvania" />
  <meta name="citation_author_orcid" content="0000-0003-0002-5761" />
  <meta name="twitter:creator" content="@dnicholson329" />
  <meta name="citation_author" content="Faisal Alquaddoomi" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado School of Medicine" />
  <meta name="citation_author_institution" content="Center for Health Artificial Intellegence (CHAI), University of Colorado School of Medicine" />
  <meta name="citation_author_orcid" content="0000-0003-4297-8747" />
  <meta name="citation_author" content="Vincent Rubinetti" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado School of Medicine" />
  <meta name="citation_author_institution" content="Center for Health Artificial Intellegence (CHAI), University of Colorado School of Medicine" />
  <meta name="citation_author" content="Casey S. Greene" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado School of Medicine" />
  <meta name="citation_author_institution" content="Center for Health Artificial Intellegence (CHAI), University of Colorado School of Medicine" />
  <meta name="citation_author_orcid" content="0000-0001-8713-9213" />
  <meta name="twitter:creator" content="@greenescientist" />
  <link rel="canonical" href="https://greenelab.github.io/word_lapse_manuscript/" />
  <meta property="og:url" content="https://greenelab.github.io/word_lapse_manuscript/" />
  <meta property="twitter:url" content="https://greenelab.github.io/word_lapse_manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://greenelab.github.io/word_lapse_manuscript/" />
  <meta name="citation_pdf_url" content="https://greenelab.github.io/word_lapse_manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://greenelab.github.io/word_lapse_manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://greenelab.github.io/word_lapse_manuscript/v/40d309ace7dd5b81a3fee79da849aa60d9ff859e/" />
  <meta name="manubot_html_url_versioned" content="https://greenelab.github.io/word_lapse_manuscript/v/40d309ace7dd5b81a3fee79da849aa60d9ff859e/" />
  <meta name="manubot_pdf_url_versioned" content="https://greenelab.github.io/word_lapse_manuscript/v/40d309ace7dd5b81a3fee79da849aa60d9ff859e/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Changing word meanings in biomedical literature reveal pandemics and new technologies</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://greenelab.github.io/word_lapse_manuscript/v/40d309ace7dd5b81a3fee79da849aa60d9ff859e/">permalink</a>)
was automatically generated
from <a href="https://github.com/greenelab/word_lapse_manuscript/tree/40d309ace7dd5b81a3fee79da849aa60d9ff859e">greenelab/word_lapse_manuscript@40d309a</a>
on July 8, 2022.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>David N. Nicholson</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0002-5761">0000-0003-0002-5761</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/danich1">danich1</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/dnicholson329">dnicholson329</a><br>
<small>
Genomics and Computational Biology Program, University of Pennsylvania
· Funded by The Gordon and Betty Moore Foundation, GBMF4552; The National Human Genome Research Institute, R01 HG010067
</small></p></li>
<li><p><strong>Faisal Alquaddoomi</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4297-8747">0000-0003-4297-8747</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/falquaddoomi">falquaddoomi</a><br>
<small>
Department of Biomedical Informatics, University of Colorado School of Medicine; Center for Health Artificial Intellegence (CHAI), University of Colorado School of Medicine
· Funded by The Gordon and Betty Moore Foundation, GBMF4552; The National Human Genome Research Institute, R01 HG010067
</small></p></li>
<li><p><strong>Vincent Rubinetti</strong><br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/vincerubinetti">vincerubinetti</a><br>
<small>
Department of Biomedical Informatics, University of Colorado School of Medicine; Center for Health Artificial Intellegence (CHAI), University of Colorado School of Medicine
· Funded by The Gordon and Betty Moore Foundation, GBMF4552; The National Human Genome Research Institute, R01 HG010067
</small></p></li>
<li><p><strong>Casey S. Greene</strong><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8713-9213">0000-0001-8713-9213</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cgreene">cgreene</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/greenescientist">greenescientist</a><br>
<small>
Department of Biomedical Informatics, University of Colorado School of Medicine; Center for Health Artificial Intellegence (CHAI), University of Colorado School of Medicine
· Funded by The Gordon and Betty Moore Foundation, GBMF4552; The National Human Genome Research Institute, R01 HG010067
</small></p>
<p>To whom correspondence should be addressed:</p>
<pre><code>CSG: casey.s.greene@cuanschutz.edu</code></pre></li>
</ul>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>While we often think of words as having a fixed meaning that we use to describe a changing world, words are also dynamic and changing.
Scientific research can also be remarkably fast-moving, with new concepts or approaches rapidly gaining mind share.
We examined scientific writing, both preprint and pre-publication peer-reviewed text, to identify terms that have changed and examine their use.
One particular challenge that we faced was that the shift from closed to open access publishing meant that the size of available corpora changed by over an order of magnitude in the last two decades.
We developed an approach to evaluate semantic shift by accounting for both intra- and inter-year variability using multiple integrated models.
Using this strategy and examining year-by-year changes revealed thousands of change points in both corpora.
We found change points for tokens including ‘cas9’, ‘pandemic’, and ‘sars’ among many others.
The consistent change-points between pre-publication peer-reviewed and preprinted text were largely related to the COVID-19 pandemic.
We developed a web app for exploration (https://greenelab.github.io/word-lapse/) that enables users to investigate individual terms.
To our knowledge, this analysis is the first to examine semantic shift in biomedical preprints and pre-publication peer-reviewed text, and it lays the foundation for future work to examine how terms acquire new meaning and the extent to which that process is encouraged or discouraged by peer review.</p>
<h1 id="introduction">Introduction</h1>
<p>Language is constantly evolving, and the meaning that we ascribe to words changes over time.
For example, the word “nice” was used to mean foolish or innocent back in the 15th-17th century; then, it underwent a positive shift to its current meaning of “pleasant or delightful”<span class="citation" data-cites="16LaV5jTS">[<a href="#ref-16LaV5jTS" role="doc-biblioref">1</a>]</span>.
These shifts occur for many reasons.
For example, writers may use new metaphors or substitute words for others with similar meanings in a process known as metonymy <span class="citation" data-cites="16LaV5jTS">[<a href="#ref-16LaV5jTS" role="doc-biblioref">1</a>]</span>.
Studying these shifts can provide a nuanced understanding of how language adapts to describe our world.</p>
<p>Scientific fields of inquiry also change, sometimes rapidly, as researchers devise and test new hypotheses and applications.
For example, the repurposing of the CRISPR-Cas9 system to a pervasive tool for genome editing has altered how we discuss molecular entities.
Microbes use this as an immune system to defend against viruses.
Scientists repurposed this system for genome editing <span class="citation" data-cites="15T32hvm6">[<a href="#ref-15T32hvm6" role="doc-biblioref">2</a>]</span>, leading to changes in the use of the term.
Science is a field with substantial written communication <span class="citation" data-cites="TQgSqYqx">[<a href="#ref-TQgSqYqx" role="doc-biblioref">3</a>]</span>, both via published papers <span class="citation" data-cites="u294RvPz">[<a href="#ref-u294RvPz" role="doc-biblioref">4</a>]</span> and preprints <span class="citation" data-cites="i9BKPuqb 1BL18WAmW">[<a href="#ref-i9BKPuqb" role="doc-biblioref">5</a>,<a href="#ref-1BL18WAmW" role="doc-biblioref">6</a>]</span>.
Examining scientific manuscripts with computational linguistics can reveal longitudinal trends in scientific research.</p>
<p>Studying changes in the use of word meanings is called semantic shift detection.
Approaches for semantic shift detection examine time series datasets that capture word usage patterns, both with respect to frequency and structure.
Typically, these time series are generated for individual words by training a unique model on text binned by a selected time period <span class="citation" data-cites="j6Vt5Hlm HFrLqOEz 18czzWopC">[<a href="#ref-j6Vt5Hlm" role="doc-biblioref">7</a>,<a href="#ref-HFrLqOEz" role="doc-biblioref">8</a>,<a href="#ref-18czzWopC" role="doc-biblioref">9</a>]</span>.
Methods are then applied to identify “change points” where a word’s meaning has changed <span class="citation" data-cites="21OOkxjt">[<a href="#ref-pAmipxbM" role="doc-biblioref">11</a>]</span>.</p>
<p>Semantic shifts have been examined in many sources.
Analysis has included newspapers <span class="citation" data-cites="lVEHA3Zy KSkyAPSb qEhBr3RQ">[<a href="#ref-lVEHA3Zy" role="doc-biblioref">12</a>,<a href="#ref-KSkyAPSb" role="doc-biblioref">13</a>,<a href="#ref-qEhBr3RQ" role="doc-biblioref">14</a>]</span>, books <span class="citation" data-cites="j6Vt5Hlm">[<a href="#ref-j6Vt5Hlm" role="doc-biblioref">7</a>]</span>, reddit <span class="citation" data-cites="FQTxXyuO">[<a href="#ref-FQTxXyuO" role="doc-biblioref">15</a>]</span>, and Twitter <span class="citation" data-cites="7gkQDvnc">[<a href="#ref-7gkQDvnc" role="doc-biblioref">16</a>]</span>.
Researchers have examined topics in information retrieval <span class="citation" data-cites="TEUTroLA">[<a href="#ref-TEUTroLA" role="doc-biblioref">17</a>]</span>, and in biomedicine COVID-19 has been examined multiple times <span class="citation" data-cites="wXphJ7n3 ynODVhC7 K1SZY72U">[<a href="#ref-wXphJ7n3" role="doc-biblioref">18</a>,<a href="#ref-ynODVhC7" role="doc-biblioref">19</a>,<a href="#ref-K1SZY72U" role="doc-biblioref">20</a>]</span>.
The amount of open access biomedical literature has dramatically increased in the last two decades, laying the groundwork for the large-scale analysis of semantic shifts in biomedicine.</p>
<p>We examine these semantic shifts in this rapidly growing body of open access text.
We include both published papers and preprints in our analysis.
We found that novel strategies integrating multiple models for each year sidestepped the challenge of instability in the machine learning models and allowed us to estimate intra- and inter-year variability.
We identify semantic change points for each token.
We examine key cases and provide the full set of research products, including change points and machine learning models, as openly licensed tools for the community.
We also created a webserver that allows users to analyze tokens of interest on the fly, examining both the most similar terms within a year and temporal trends.</p>
<h1 id="methods">Methods</h1>
<h2 id="biomedical-corpora-examined">Biomedical Corpora Examined</h2>
<h3 id="pubtator-central">Pubtator Central</h3>
<p>Pubtator Central is an open-access resource containing annotated abstracts and full-text annotated with entity recognition systems for biomedical concepts <span class="citation" data-cites="18ZyyTcTe">[<a href="#ref-18ZyyTcTe" role="doc-biblioref">21</a>]</span>.
The methods used are TaggerOne <span class="citation" data-cites="11YUuHulp">[<a href="#ref-11YUuHulp" role="doc-biblioref">22</a>]</span> to tag diseases, chemicals, and cell line entities, GNormPlus <span class="citation" data-cites="aOPX10e0">[<a href="#ref-aOPX10e0" role="doc-biblioref">23</a>]</span> to tag
genes, SR4GN <span class="citation" data-cites="NXIFrudx">[<a href="#ref-NXIFrudx" role="doc-biblioref">24</a>]</span> to tag species, and tmVar <span class="citation" data-cites="17LQKv7vO">[<a href="#ref-17LQKv7vO" role="doc-biblioref">25</a>]</span> to tag genetic mutations.
We initially downloaded this resource on December 07th, 2021, and processed over 30 million documents.
This resource contains documents that date back to the pre-1800s to the year 2021; however, due to the low sample size in early years, we only used documents published from 2000 to 2021.
The resource was subsequently updated with documents from 2021.
We also downloaded a later version on March 09th, 2022, and merged both versions using each document’s doc_id field to produce the corpus used in this analysis.
We divided documents by publication year and then preprocessed each using spacy’s en_core_web_sm model <span class="citation" data-cites="q2fFAZTG">[<a href="#ref-q2fFAZTG" role="doc-biblioref">26</a>]</span>.
We replaced each tagged word or phrase with its corresponding entity type and entity id for every sentence that contained an annotation.
Then, we used spacy to break sentences into individual tokens and normalized each token to its root form via lemmatization.
After preprocessing, we used every sentence to train multiple natural language models designed to represent words based on their context.</p>
<h3 id="biomedical-preprints">Biomedical Preprints</h3>
<p>BioRxiv <span class="citation" data-cites="i9BKPuqb">[<a href="#ref-i9BKPuqb" role="doc-biblioref">5</a>]</span> and MedRxiv <span class="citation" data-cites="1BL18WAmW">[<a href="#ref-1BL18WAmW" role="doc-biblioref">6</a>]</span> are repositories that contain preprints for the life science community.
MedRxiv mainly focuses on preprints that mention patient research, while bioRxiv focuses on general biology.
We downloaded a snapshot of both resources on March 4th, 2022, using their respective Amazon S3 bucket <span class="citation" data-cites="9gVmEi52 v3H8juk7">[<a href="#ref-9gVmEi52" role="doc-biblioref">27</a>,<a href="#ref-v3H8juk7" role="doc-biblioref">28</a>]</span>.
This snapshot contained 172,868 BioRxiv preprints and 37,517 MedRxiv preprints.
These resources allow authors to post multiple versions of a single preprint.
To prevent duplication bias, we filtered every preprint to its most recent version and sorted each preprint into its respective posted year.
Unlike Pubtator Central, these filtered preprints do not contain any annotations.
Therefore, we used TaggerOne <span class="citation" data-cites="11YUuHulp">[<a href="#ref-11YUuHulp" role="doc-biblioref">22</a>]</span> to tag every chemical and disease entity and GNormplus <span class="citation" data-cites="aOPX10e0">[<a href="#ref-aOPX10e0" role="doc-biblioref">23</a>]</span> to tag every gene and species entity for our preprint set.
Once tagged, we used spacy to preprocess every preprint as described in our Pubtator Central section.</p>
<h2 id="constructing-word-embeddings-for-semantic-change-detection">Constructing Word Embeddings for Semantic Change Detection</h2>
<p>Word2vec <span class="citation" data-cites="1GhHIDxuW">[<a href="#ref-1GhHIDxuW" role="doc-biblioref">29</a>]</span> is a natural language processing model designed to model words based on their respective neighbors in the form of dense vectors.
This suite of models comes in two forms, a skipgram model and a continuous bags of words (CBOW) model.
The skipgram model generates these vectors by having a shallow neural network predict a word’s neighbors given the word, while the CBOW model predicts the word given its neighbors.
We used the CBOW model to construct word vectors for each year.
Despite the power of these word2vec models, these models are known to differ both due to randomization within year and year-to-year variability across years <span class="citation" data-cites="18rNo4lJI 8G9bCV2k ImhJ75vq bXXIQj3W">[<a href="#ref-18rNo4lJI" role="doc-biblioref">30</a>,<a href="#ref-8G9bCV2k" role="doc-biblioref">31</a>,<a href="#ref-ImhJ75vq" role="doc-biblioref">32</a>,<a href="#ref-bXXIQj3W" role="doc-biblioref">33</a>]</span>.
To control for run-to-run variability, we examined both intra-year and inter-year relationships.
Each year, we trained ten different CBOW models using the following parameters: vector size of 300, 10 epochs, minimum frequency cutoff of 5, and a window size of 16 for abstracts.
Every model has its own unique vector space following training, making it difficult to compare two models without a correction step.
We used orthogonal Procrustes <span class="citation" data-cites="NjbU71da">[<a href="#ref-NjbU71da" role="doc-biblioref">34</a>]</span> to align models.
We aligned all trained CBOW models for the Pubtator Central dataset to the first model trained in 2021.
Likewise, we aligned all CBOW models for the BioRxiv/MedRxiv dataset to the first model trained in 2021.
We used UMAP <span class="citation" data-cites="157h5hA34">[<a href="#ref-157h5hA34" role="doc-biblioref">35</a>]</span> to visually examine the aligned models.
We trained this model using the following parameters: cosine distance metric, random_state of 100, 25 for n_neighbors, a minimum distance of 0.99, and 50 n_epochs.</p>
<h2 id="detecting-semantic-changes-across-time">Detecting semantic changes across time</h2>
<p>Once word2vec models are aligned, the next step is to detect semantic change.<br />
Semantic change events are often detected through time series analysis <span class="citation" data-cites="TB6Pquc2">[<a href="#ref-TB6Pquc2" role="doc-biblioref">36</a>]</span>.
We constructed a time series sequence for every token by calculating its distance within a given year (intra-year) and across each year (inter-year).
We used the model pairs constructed from the same year to calculate an intra-year distance.
Then, we calculated the cosine distance between each token and its corresponding counterpart for every generated pair.
Cosine distance is a metric bounded between zero and two, where a score of zero means two vectors are the same, and a score of two means both vectors are different.
For the inter-year distance, we used the Cartesian product of every model between two years and calculated the distance between tokens in the same way as the intra-year distance.
Following both calculations, we combined both metrics by taking the ratio of the average inter-year distance over the average intra-year distance.
Through this approach, tokens with high intra-year instability will be penalized and vice-verse for more stable tokens.
Along with token distance calculations, it has been shown that including token frequency improves results compared to using distance alone <span class="citation" data-cites="1BaA3rWCK">[<a href="#ref-1BaA3rWCK" role="doc-biblioref">37</a>]</span>.
We calculated token frequency as the ratio of token frequency in the more recent year over the frequency of the previous year.
Then, we combined both the frequency and distance ratios to make the final metric.</p>
<p>Following time series construction, we performed change point detection, which is a process that uses statistical techniques to detect abnormalities within a given time series.
We used the CUSUM algorithm <span class="citation" data-cites="pAmipxbM">[<a href="#ref-pAmipxbM" role="doc-biblioref">11</a>]</span> to detect these abnormalities.
This algorithm uses a rolling sum of the differences between two timepoints and checks whether the sum is greater than a threshold.
A changepoint is considered to have occurred if the sum is greater than a threshold.
We used the 99th percentile on every generated timepoint as the threshold.
Then, we ran the CUSUM algorithm using a drift of 0 and default settings for all other parameters.</p>
<h1 id="results">Results</h1>
<h2 id="models-can-be-aligned-and-compared-within-and-between-years">Models can be aligned and compared within and between years</h2>
<p>We examined how the usage of tokens in biomedical text changes over time.
Our evaluation was derived from machine learning models designed to predict the actual token given a portion of its surrounding tokens.
Each token was represented as a vector in a coordinate space constructed by these models.
However, training these models is stochastic, which results in arbitrary coordinate spaces.
Model alignment is an essential step in allowing word2vec models to be compared <span class="citation" data-cites="xQaNPZgV 1346YOPVE">[<a href="#ref-xQaNPZgV" role="doc-biblioref">38</a>,<a href="#ref-1346YOPVE" role="doc-biblioref">39</a>]</span>.
Before alignment, each model has its own unique coordinate space (Figures <a href="#fig:word2vec_alignment">1</a>A), and each word is represented within that space (Figure <a href="#fig:word2vec_alignment">1</a>B).
Alignment projects every model onto a shared coordinate space (Figure <a href="#fig:word2vec_alignment">1</a>C), enabling direct token comparison.
We randomly selected 100 tokens to confirm that alignment worked as expected.
In aligned models, tokens in the global spcae were more similar to themselves within year than between years, while identical tokens in unaligned models were completely distinct (Figure <a href="#fig:word2vec_alignment">1</a>D).
Local distances were unaffected by alignment (Figure <a href="#fig:word2vec_alignment">1</a>D), as token-neighbor distances were unaffected by the alignment procedure.</p>
<div id="fig:word2vec_alignment" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/danich1/biovectors/1c2f91cd05ed6714525f54a006f9a78bf7fe2d31/figure_generation/output/Figure_1.png" style="width:100.0%" alt="Figure 1: A. Without alignment, each word2vec model has its own coordinate space. This is a UMAP visualization of 5000 randomly sampled tokens from 5 distinct Word2Vec models trained on the text published in 2010. Each data point represents a token, and the color represents the respective Word2Vec model. B. The highlighted token ‘probiotics’ shows up in its respective clusters. Each data point represents a token, and the color represents the Word2Vec model. C. After the alignment step, the token ‘probiotic’ is closer in vector space. Each data point represents a token, and the color represents the different Word2Vec models. D. In the global coordinate space, token distances appear to be vastly different without alignment, but become closer upon alignment, while local distances, evaluated using neighbors, are unaffected. This boxplot shows the average distance of 100 randomly sampled tokens shared in every year from 2000 to 2021. The x-axis shows the various groups being compared (tokens against themselves via intra-year and inter-year distances and tokens against their corresponding neighbors. The y axis shows the averaged distance for every year." />
<figcaption aria-hidden="true"><span>Figure 1:</span> A. Without alignment, each word2vec model has its own coordinate space.
This is a UMAP visualization of 5000 randomly sampled tokens from 5 distinct Word2Vec models trained on the text published in 2010.
Each data point represents a token, and the color represents the respective Word2Vec model.
B. The highlighted token ‘probiotics’ shows up in its respective clusters.
Each data point represents a token, and the color represents the Word2Vec model.
C. After the alignment step, the token ‘probiotic’ is closer in vector space.
Each data point represents a token, and the color represents the different Word2Vec models.
D. In the global coordinate space, token distances appear to be vastly different without alignment, but become closer upon alignment, while local distances, evaluated using neighbors, are unaffected.
This boxplot shows the average distance of 100 randomly sampled tokens shared in every year from 2000 to 2021.
The x-axis shows the various groups being compared (tokens against themselves via intra-year and inter-year distances and tokens against their corresponding neighbors.
The y axis shows the averaged distance for every year.</figcaption>
</figure>
</div>
<p>The landscape of biomedical publishing has changed rapidly during the period of our dataset.
The texts for our analysis were open access manuscripts available through PubMed Central.
The growth in the amount of available text and the uneven adoption of open access publishing during the interval studied was expected to induce changes in the underlying machine learning models, making comparisons more difficult.
We found that the number of tokens available for model building, i.e., those in PMC OA, increased dramatically during this time (Figure <a href="#fig:novel_distance_validation">2</a>A).
This was expected to create a pattern where models trained in earlier years were more variable than those from later years simply due to the limited sample size in early years.
We aimed to correct for this change in the underlying models by developing a statistic that, instead of using pairwise comparisons of token distances between individual models, integrated multiple models for each year by comparing tokens’ intra- and inter-year variabilities.
We defined the statistic as the ratio of the average distance between two years over the sum of the average distance within each year respectively.</p>
<div id="fig:novel_distance_validation" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/danich1/biovectors/b48e44d1a06106d3a76b35980384bc0c0c2cfaf9/figure_generation/output/Figure_2.png" style="width:100.0%" alt="Figure 2: A. The number of tokens our models have trained on increases over time. This line plot shows the number of unique tokens seen by our various machine learning models. The x-axis depicts the year and the y-axis shows the token count. B. Earlier years compared to 2010 have greater distances than later years. This confidence interval plot shows the collective distances obtained by sampling 100 tokens that are present from every year using a single model approach. The x-axis shows a given year and the y-axis shows the distance metric. C. Later years have a lower intra-distance variability compared to the earlier years. This confidence interval plot shows the collective distances obtained by sampling 100 tokens that are present from every year using our multi-model approach. The x-axis shows a given year and the y-axis shows the distance metric." />
<figcaption aria-hidden="true"><span>Figure 2:</span> A. The number of tokens our models have trained on increases over time.
This line plot shows the number of unique tokens seen by our various machine learning models.
The x-axis depicts the year and the y-axis shows the token count.
B. Earlier years compared to 2010 have greater distances than later years.
This confidence interval plot shows the collective distances obtained by sampling 100 tokens that are present from every year using a single model approach.
The x-axis shows a given year and the y-axis shows the distance metric.
C. Later years have a lower intra-distance variability compared to the earlier years.
This confidence interval plot shows the collective distances obtained by sampling 100 tokens that are present from every year using our multi-model approach.
The x-axis shows a given year and the y-axis shows the distance metric.</figcaption>
</figure>
</div>
<p>We expected most tokens to undergo minor changes from year to year, while substantial changes likely suggested model drift as opposed to true linguistic change.
We measured the extent to which tokens differed from themselves using the standard single-model approach and our integrated statistic.
We filtered the token list to only contain tokens present in every year and compared their distance to the midpoint year, 2010, using the single-model and integrated-models strategies.
We found that distances tended were markedly larger in the earliest years, where we expected models to be least stable, using the traditional approach (Figure <a href="#fig:novel_distance_validation">2</a>B).
The integrated model approach did not display the same pattern in the earliest years (Figure <a href="#fig:novel_distance_validation">2</a>C).
Both trends reinforce that training on smaller corpora will lead to high variation and that an integrated model strategy is needed <span class="citation" data-cites="ImhJ75vq">[<a href="#ref-ImhJ75vq" role="doc-biblioref">32</a>]</span>.
Based on these results, we used the integrated-model strategy to calculate inter-year token distances for the remainder of this work.</p>
<h2 id="terms-exhibit-detectable-changes-in-usage">Terms exhibit detectable changes in usage</h2>
<div id="fig:preprint_published_changepoints" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/danich1/biovectors/576df207026d0573fbbb00b95c264aea0bd8174b/figure_generation/output/Figure_3.png" style="width:100.0%" alt="Figure 3: A. The number of change points increases over time in PMCOA. The x-axis shows the various time periods, while the y-axis depicts the number of detected change points. B. Regarding preprints, the greatest number of change points was during 2018-2019. The x-axis shows the various time periods, while the y-axis depicts the number of detected change points. C. The token ‘cas9’ was detected to have a change point at 2012-2013. The x-axis shows the time period since the first appearance of the token, and the y-axis shows the change metric. D. ‘sars’ has two detected change points within the PMCOA corpus. The x-axis shows the time period since the first appearance of the token, and the y-axis shows the change metric." />
<figcaption aria-hidden="true"><span>Figure 3:</span> A. The number of change points increases over time in PMCOA.
The x-axis shows the various time periods, while the y-axis depicts the number of detected change points.
B. Regarding preprints, the greatest number of change points was during 2018-2019.
The x-axis shows the various time periods, while the y-axis depicts the number of detected change points.
C. The token ‘cas9’ was detected to have a change point at 2012-2013.
The x-axis shows the time period since the first appearance of the token, and the y-axis shows the change metric.
D. ‘sars’ has two detected change points within the PMCOA corpus.
The x-axis shows the time period since the first appearance of the token, and the y-axis shows the change metric.</figcaption>
</figure>
</div>
<p>We next sought to identify tokens that changed during the 2000-2021 interval for the text from PubMed Central’s Open Access Corpus (PMCOA) and the 2015-2022 interval for our preprint corpus.
We performed change point detection using the CUSUM algorithm with distances calculated with the integrated-model approach to correct for systematic differences in the underlying corpora.
We found 41281 terms with a detected change point from PMCOA and 2266 terms from preprints (Figures <a href="#fig:preprint_published_changepoints">3</a>A and <a href="#fig:preprint_published_changepoints">3</a>B), and the vast majority (38019 for PMCOA and 2260 for preprints) had just a single change-point.</p>
<p>We explored individual change points.
We detected one in PMCOA for ‘cas9’ from 2012 to 2013 (Figure <a href="#fig:preprint_published_changepoints">3</a>C).
Before the change point, its closest neighbors were related genetic elements (e.g., ‘cas’1-3).
After the change point, its closest neighbors became terms related to targeting, sgRNA, and gRNA, as well as other genome editing strategies, ’talen’ and ‘zfns’ (Table <a href="#tbl:cas9_neighbor_table">1</a>).
For some terms, we detected multiple change points within the studied interval.
We detected change points for ‘SARS’ from 2002 to 2003 and 2019 to 2020 (Figure <a href="#fig:preprint_published_changepoints">3</a>D), consistent with the emergences of SARS-CoV <span class="citation" data-cites="YuxGez24">[<a href="#ref-YuxGez24" role="doc-biblioref">40</a>]</span> and SARS-CoV-2 <span class="citation" data-cites="1ExALQ5eN vvMItrwz">[<a href="#ref-1ExALQ5eN" role="doc-biblioref">41</a>,<a href="#ref-vvMItrwz" role="doc-biblioref">42</a>]</span> as observed human pathogens.
We found miscellaneous neighbors before each change point, with use consistent with the acronym for Severe Acute Respiratory Syndrome after each (Tables <a href="#tbl:sars_neighbor_table_one">2</a> and <a href="#tbl:sars_neighbor_table_two">3</a>).</p>
<p>Out of all change points, we observed 200 tokens with at least one change point in each corpus.
Only 25 of the 200 terms were detected to have simultaneous changes between the preprint and PMCOA corpora.
We examined the overlap of detected change points between preprints and published articles.
Many of these 25 were related to the COVID-19 pandemic (Supplementary Table <a href="#tbl:published_preprint_change_table">S1</a>).
The complete set of detected change points is available for further analysis (see Data Availability and Software).</p>
<div id="tbl:cas9_neighbor_table" class="tablenos">
<table id="tbl:cas9_neighbor_table">
<caption><span>Table 1:</span> The fifteen most similar neighbors to the token ‘cas9’ for the years 2012 and 2013. </caption>
<thead>
<tr class="header">
<th>2012</th>
<th>2013</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>cas2</td>
<td>sgrna</td>
</tr>
<tr class="even">
<td>crispr1</td>
<td>talen</td>
</tr>
<tr class="odd">
<td>cas3</td>
<td>spcas9</td>
</tr>
<tr class="even">
<td>cas1</td>
<td>zfns</td>
</tr>
<tr class="odd">
<td>cas10</td>
<td>grna</td>
</tr>
<tr class="even">
<td>crispr3</td>
<td>zfn</td>
</tr>
<tr class="odd">
<td>tracrrna</td>
<td>dcas9</td>
</tr>
<tr class="even">
<td>crispr</td>
<td>nickase</td>
</tr>
<tr class="odd">
<td>csn1</td>
<td>pcocas9</td>
</tr>
<tr class="even">
<td>crispr4</td>
<td>crispr</td>
</tr>
<tr class="odd">
<td>cas7</td>
<td>sgrnas</td>
</tr>
<tr class="even">
<td>cas6e</td>
<td>meganuclease</td>
</tr>
<tr class="odd">
<td>cas4</td>
<td>tracrrna</td>
</tr>
<tr class="even">
<td>cse1</td>
<td>crispri</td>
</tr>
<tr class="odd">
<td>cas6</td>
<td>crrna</td>
</tr>
</tbody>
</table>
</div>
<div id="tbl:sars_neighbor_table_one" class="tablenos">
<table id="tbl:sars_neighbor_table_one">
<caption><span>Table 2:</span> The fifteen most similar neighbors to the token ‘sars’ for the years 2002 and 2003. </caption>
<colgroup>
<col style="width: 26%" />
<col style="width: 73%" />
</colgroup>
<thead>
<tr class="header">
<th>2002</th>
<th>2003</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>qsar</td>
<td>species_227859</td>
</tr>
<tr class="even">
<td>herbicidal</td>
<td>mesh_c000657245</td>
</tr>
<tr class="odd">
<td>antiplasmodial</td>
<td>severe acute respiratory syndrome-related coronavirus (species_694009)</td>
</tr>
<tr class="even">
<td>arylpiperazine</td>
<td>unidentified human coronavirus (species_694448)</td>
</tr>
<tr class="odd">
<td>a]pyridine</td>
<td>SARS1 (gene_6301)</td>
</tr>
<tr class="even">
<td>leishmanicidal</td>
<td>ebola virus sp. (species_205488)</td>
</tr>
<tr class="odd">
<td>naphthyridine</td>
<td>pandemic</td>
</tr>
<tr class="even">
<td>indolo[2,1</td>
<td>coronavirus infections (mesh_d018352)</td>
</tr>
<tr class="odd">
<td>b]quinazoline-6,12</td>
<td>coronavirus</td>
</tr>
<tr class="even">
<td>nematocidal</td>
<td>ebola virus (species_1570291)</td>
</tr>
<tr class="odd">
<td>f]isoxazolo[2,3</td>
<td>severe acute respiratory syndrome (mesh_d045169)</td>
</tr>
<tr class="even">
<td>5-(4</td>
<td>paramyxovirus</td>
</tr>
<tr class="odd">
<td>cholinephosphotransferase</td>
<td>viruse</td>
</tr>
<tr class="even">
<td>oxovanadium(iv</td>
<td>drosten</td>
</tr>
<tr class="odd">
<td>catecholase</td>
<td>virologist</td>
</tr>
</tbody>
</table>
</div>
<div id="tbl:sars_neighbor_table_two" class="tablenos">
<table id="tbl:sars_neighbor_table_two">
<caption><span>Table 3:</span> The fifteen most similar neighbors to the token ‘sars’ for the years 2002 and 2003. </caption>
<colgroup>
<col style="width: 26%" />
<col style="width: 73%" />
</colgroup>
<thead>
<tr class="header">
<th>2019</th>
<th>2020</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>g.o.</td>
<td>sar</td>
</tr>
<tr class="even">
<td>nsp13</td>
<td>mers</td>
</tr>
<tr class="odd">
<td>40/367</td>
<td>cov</td>
</tr>
<tr class="even">
<td>lissodendoryx</td>
<td>sars-1</td>
</tr>
<tr class="odd">
<td>lutken</td>
<td>severe acute respiratory syndrome-related coronavirus (species_694009)</td>
</tr>
<tr class="even">
<td>sarr</td>
<td>coronaviruse</td>
</tr>
<tr class="odd">
<td>sar</td>
<td>middle east respiratory syndrome-related coronavirus (species_1335626)</td>
</tr>
<tr class="even">
<td>ophiura ophiura (species_72673)</td>
<td>cov.</td>
</tr>
<tr class="odd">
<td>verrill</td>
<td>coronavirus infections (mesh_d018352)</td>
</tr>
<tr class="even">
<td>hirondelle</td>
<td>mers-</td>
</tr>
<tr class="odd">
<td>kobelt</td>
<td>covs</td>
</tr>
<tr class="even">
<td>azorean</td>
<td>severe acute respiratory syndrome coronavirus 2 (species_2697049)</td>
</tr>
<tr class="odd">
<td>rusby</td>
<td>severe acute respiratory syndrome (mesh_d045169)</td>
</tr>
<tr class="even">
<td>d’orbigny</td>
<td>sarscov</td>
</tr>
<tr class="odd">
<td>psychropotes longicauda (species_55639)</td>
<td>sarscov-2</td>
</tr>
</tbody>
</table>
</div>
<h2 id="the-word-lapse-application-is-an-online-resource-for-manual-examination-of-biomedical-tokens">The word-lapse application is an online resource for manual examination of biomedical tokens</h2>
<div id="fig:website_walkthrough" class="fignos">
<figure>
<img src="images/Figure_4.png" style="width:100.0%" alt="Figure 4: A. The trajectory visualization of the token ‘pandemic’ through time. It starts at the first mention of the token and progresses through each subsequent year. Every data point shows the top five neighbors for the respective token. B. The usage frequency of the token ‘pandemic’ through time. The x-axis shows the year, and the y-axis shows the frequency for each token. C. A word cloud visualization for the top 25 neighbors for the token ‘pandemic’ each year. This visualization highlights each neighbor from a particular year and allows for the comparison between two years. Tokens in purple are shared within both years, while tokens in red or blue are unique to their respective year." />
<figcaption aria-hidden="true"><span>Figure 4:</span> A. The trajectory visualization of the token ‘pandemic’ through time.
It starts at the first mention of the token and progresses through each subsequent year.
Every data point shows the top five neighbors for the respective token.
B. The usage frequency of the token ‘pandemic’ through time.
The x-axis shows the year, and the y-axis shows the frequency for each token.
C. A word cloud visualization for the top 25 neighbors for the token ‘pandemic’ each year.
This visualization highlights each neighbor from a particular year and allows for the comparison between two years.
Tokens in purple are shared within both years, while tokens in red or blue are unique to their respective year.</figcaption>
</figure>
</div>
<p>We constructed an online application that allows users to examine how tokens change through time.
The application supports token input as text strings or as MeSH IDs, Entrez Gene IDs, and Taxonomy IDs.
Users might elect to explore the term ‘pandemic’, for which we detected a change point between 2019 and 2020.
Users can examine the token’s nearest neighbors through time (Figure <a href="#fig:website_walkthrough">4</a>A).
Using the token ‘pandemic’ as an example, users can observe that ‘epidemic’ remains similar through time, but taxid:114727 (the H1N1 subtype of influenza) only entered the nearest neighbors with the swine flu pandemic in 2009 and that MeSH:C000657245 (COVID-19) appears in 2020.
The application also shows a frequency chart depicting how often the particular token is used each year (Figure <a href="#fig:website_walkthrough">4</a>B), which can be displayed as a raw count or adjusted by the total size of the corpus.
When change points are detected, they are indicated on this panel (Figure <a href="#fig:website_walkthrough">4</a>B).
The final visualization shows the union of the nearest 25 neighbors from each year ordered by the number of years that neighbor was present (Figure <a href="#fig:website_walkthrough">4</a>C).
This visualization has a comparison function that allows users to examine differences between years.
All functionalities are fully supported across the PMCOA and preprint corpora, and users can toggle between the two.</p>
<h1 id="discussion-and-conclusion">Discussion and Conclusion</h1>
<p>Language is rapidly evolving, and the usage of words changes over time with words assimilating new meanings or associations <span class="citation" data-cites="16LaV5jTS">[<a href="#ref-16LaV5jTS" role="doc-biblioref">1</a>]</span>.
Some efforts have been made to study semantic change using biomedical text <span class="citation" data-cites="wXphJ7n3 ynODVhC7 K1SZY72U">[<a href="#ref-wXphJ7n3" role="doc-biblioref">18</a>,<a href="#ref-ynODVhC7" role="doc-biblioref">19</a>,<a href="#ref-K1SZY72U" role="doc-biblioref">20</a>]</span>; however, no such work has examined the changes evident in both pre-publication peer-reviewed and preprinted biomedical text.
We examined semantic change in both open-access biomedical corpora from Pubmed (PMCOA) and bioRxiv/MedRxiv for the 2000-2021 interval.
Studies like this one have only become feasible recently with the rapid uptake of open access publishing.
To be able to span two decades with very different data availability, we developed a novel statistic that incorporated multiple models using both inter- and intra-year distances.
Without the correction, comparing between stable and unstable models is challenging as previously reported <span class="citation" data-cites="ImhJ75vq M1LzK6yf">[<a href="#ref-ImhJ75vq" role="doc-biblioref">32</a>,<a href="#ref-M1LzK6yf" role="doc-biblioref">43</a>]</span>.
Our analysis revealed more than 41,000 different change points, including tokens such as ‘cas9’, ‘pandemic’, and ‘sars’.
Many change points overlapping between PMCOA and preprints were related to COVID-19, indicating that the COVID-19 pandemic has been strong and immediate enough to induce rapid semantic change across both publishing paradigms.</p>
<p>As the amount of preprinted text grows, future work may be able to determine the consistency and time-lag of semantic change between preprint and pre-publication peer-reviewed text - potentially predicting future change in pre-publication peer-reviewed text.
We developed a web application to enable users to investigate individual tokens - automatic approaches that use orthogonal metrics to estimate change point validity would make analysis at scale more straightforward.
Furthermore, including corpora such as the arXiv <span class="citation" data-cites="ZTyQYlDz">[<a href="#ref-ZTyQYlDz" role="doc-biblioref">44</a>]</span> or psyArXiv <span class="citation" data-cites="1D4Cz1GNr">[<a href="#ref-1D4Cz1GNr" role="doc-biblioref">45</a>]</span> repositories may reveal consistencies across a broader swath of fields or within-field analyses may reveal the earliest starting points of semantic changes that ultimately sweep through biomedicine.</p>
<h2 id="data-availability-and-software">Data Availability and Software</h2>
<p>An online version of this manuscript is available under a Creative Commons Attribution License at <a href="https://greenelab.github.io/word_lapse_manuscript/">https://greenelab.github.io/word_lapse_manuscript/</a>.
The source for the research portions of this project is licensed under the BSD-2-Clause Plus Patent at <a href="https://github.com/greenelab/biovectors">https://github.com/greenelab/biovectors</a>.
Our Word Lapse website can be found at <a href="https://greenelab.github.io/word-lapse">https://greenelab.github.io/word-lapse</a>, and the code for the website is available under a BSD-3 Clause at <a href="https://github.com/greenelab/word-lapse">https://github.com/greenelab/word-lapse</a>.
Full-text access for the bioRxiv repository is available at <a href="https://www.biorxiv.org/tdm">https://www.biorxiv.org/tdm</a>.
Full-text access for the medRxiv repository is available at <a href="https://www.medrxiv.org/tdm">https://www.medrxiv.org/tdm</a>.
Access to Pubtator Central’s Open Access subset is available on NCBI’s FTP server at <a href="https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTatorCentral/">https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTatorCentral/</a>.</p>
<h1 id="acknowledgments">Acknowledgments</h1>
<p>This work was supported by the Gordon and Betty Moore Foundation under award GBMF4552 and the National Institutes of Health’s National Human Genome Research Institute under award R01 HG010067 to CSG.
The funders had no role in the study design, data collection, and analysis, decision to publish, or preparation of the manuscript.</p>
<h1 id="competing-interest">Competing Interest</h1>
<p>DNN performed this work as a Ph.D. student at the University of Pennsylvania.
He is now employed by Digital Science.
The authors report no other competing interests for this paper.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-16LaV5jTS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Semantic Change</strong> <div class="csl-block">Elizabeth Closs Traugott</div> <em>Oxford Research Encyclopedia of Linguistics</em> (2017-03-29) <a href="https://doi.org/gp574c">https://doi.org/gp574c</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/acrefore/9780199384655.013.323">10.1093/acrefore/9780199384655.013.323</a></div></div>
</div>
<div id="ref-15T32hvm6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>A Programmable Dual-RNA–Guided DNA Endonuclease in Adaptive Bacterial Immunity</strong> <div class="csl-block">Martin Jinek, Krzysztof Chylinski, Ines Fonfara, Michael Hauer, Jennifer A Doudna, Emmanuelle Charpentier</div> <em>Science</em> (2012-08-17) <a href="https://doi.org/f22dgd">https://doi.org/f22dgd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.1225829">10.1126/science.1225829</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22745249">22745249</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6286148">PMC6286148</a></div></div>
</div>
<div id="ref-TQgSqYqx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Scientific communication pathways: an overview and introduction to a symposium</strong> <div class="csl-block">David F Zaye, WV Metanomski</div> <em>Journal of Chemical Information and Computer Sciences</em> (1986-05-01) <a href="https://doi.org/bwsxhg">https://doi.org/bwsxhg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1021/ci00050a001">10.1021/ci00050a001</a></div></div>
</div>
<div id="ref-u294RvPz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>PubMed Central: The GenBank of the published literature</strong> <div class="csl-block">Richard J Roberts</div> <em>Proceedings of the National Academy of Sciences</em> (2001-01-09) <a href="https://doi.org/bbn9k8">https://doi.org/bbn9k8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.98.2.381">10.1073/pnas.98.2.381</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/11209037">11209037</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33354">PMC33354</a></div></div>
</div>
<div id="ref-i9BKPuqb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>bioRxiv: the preprint server for biology</strong> <div class="csl-block">Richard Sever, Ted Roeder, Samantha Hindle, Linda Sussman, Kevin-John Black, Janet Argentine, Wayne Manos, John R Inglis</div> <em>Cold Spring Harbor Laboratory</em> (2019-11-06) <a href="https://doi.org/ggc46z">https://doi.org/ggc46z</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/833400">10.1101/833400</a></div></div>
</div>
<div id="ref-1BL18WAmW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>Medical preprint server debuts</strong> <div class="csl-block">Jocelyn Kaiser</div> <em>Science</em> (2019-06-05) <a href="https://doi.org/gpxkkf">https://doi.org/gpxkkf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.aay2933">10.1126/science.aay2933</a></div></div>
</div>
<div id="ref-j6Vt5Hlm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Diachronic Word Embeddings Reveal Statistical Laws of Semantic Change</strong> <div class="csl-block">William L Hamilton, Jure Leskovec, Dan Jurafsky</div> <em>arXiv</em> (2018-10-26) <a href="https://arxiv.org/abs/1605.09096">https://arxiv.org/abs/1605.09096</a></div>
</div>
<div id="ref-HFrLqOEz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>Multi-Label Zero-Shot Learning via Concept Embedding</strong> <div class="csl-block">Ubai Sandouk, Ke Chen</div> <em>arXiv</em> <a href="https://arxiv.org/abs/1606.00282">https://arxiv.org/abs/1606.00282</a></div>
</div>
<div id="ref-18czzWopC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Deep Neural Models of Semantic Shift</strong> <div class="csl-block">Alex Rosenfeld, Katrin Erk</div> <em>Proceedings of the 2018 Conference of the North American Chapter of
the Association for Computational Linguistics: Human Language
Technologies, Volume 1 (Long Papers)</em> (2018) <a href="https://doi.org/gp574f">https://doi.org/gp574f</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18653/v1/n18-1044">10.18653/v1/n18-1044</a></div></div>
</div>
<div id="ref-21OOkxjt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>Bayesian Online Changepoint Detection</strong> <div class="csl-block">Ryan Prescott Adams, David JC MacKay</div> <em>arXiv</em> (2007-10-22) <a href="https://arxiv.org/abs/0710.3742">https://arxiv.org/abs/0710.3742</a></div>
</div>
<div id="ref-pAmipxbM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Adaptive filtering and change detection</strong> <div class="csl-block">Fredrik Gustafsson, Fredrik Gustafsson</div> <em>Citeseer</em> (2000)</div>
</div>
<div id="ref-lVEHA3Zy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Tracing armed conflicts with diachronic word embedding models</strong> <div class="csl-block">Andrey Kutuzov, Erik Velldal, Lilja Øvrelid</div> <em>Proceedings of the Events and Stories in the News Workshop</em> (2017) <a href="https://doi.org/ghx5gj">https://doi.org/ghx5gj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18653/v1/w17-2705">10.18653/v1/w17-2705</a></div></div>
</div>
<div id="ref-KSkyAPSb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>Words are Malleable: Computing Semantic Shifts in Political and Media Discourse</strong> <div class="csl-block">Hosein Azarbonyad, Mostafa Dehghani, Kaspar Beelen, Alexandra Arkut, Maarten Marx, Jaap Kamps</div> <em>arXiv</em> (2017-11-16) <a href="https://arxiv.org/abs/1711.05603">https://arxiv.org/abs/1711.05603</a></div>
</div>
<div id="ref-qEhBr3RQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>Reading Between the Lines: Prediction of Political Violence Using Newspaper Text</strong> <div class="csl-block">HANNES MUELLER, CHRISTOPHER RAUH</div> <em>American Political Science Review</em> (2017-12-14) <a href="https://doi.org/gdj77d">https://doi.org/gdj77d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1017/s0003055417000570">10.1017/s0003055417000570</a></div></div>
</div>
<div id="ref-FQTxXyuO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Detection of Emerging Drugs Involved in Overdose via Diachronic Word Embeddings of Substances Discussed on Social Media</strong> <div class="csl-block">Austin P Wright, Christopher M Jones, Duen Horng Chau, R Matthew Gladden, Steven A Sumner</div> <em>Journal of Biomedical Informatics</em> (2021-05) <a href="https://doi.org/gp8zph">https://doi.org/gp8zph</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.jbi.2021.103824">10.1016/j.jbi.2021.103824</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34048933">34048933</a></div></div>
</div>
<div id="ref-7gkQDvnc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Statistically Significant Detection of Linguistic Change</strong> <div class="csl-block">Vivek Kulkarni, Rami Al-Rfou, Bryan Perozzi, Steven Skiena</div> <em>arXiv</em> (2014-11-13) <a href="https://arxiv.org/abs/1411.3315">https://arxiv.org/abs/1411.3315</a></div>
</div>
<div id="ref-TEUTroLA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Semantic word shifts in a scientific domain</strong> <div class="csl-block">Baitong Chen, Ying Ding, Feicheng Ma</div> <em>Scientometrics</em> (2018-07-13) <a href="https://doi.org/gd7bd7">https://doi.org/gd7bd7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s11192-018-2843-2">10.1007/s11192-018-2843-2</a></div></div>
</div>
<div id="ref-wXphJ7n3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Semantic Changepoint Detection for Finding Potentially Novel Research Publications</strong> <div class="csl-block">Bhavish Dinakar, Mayla R Boguslav, Carsten Görg, Deendayal Dinakarpandian</div> <em>Biocomputing 2021</em> (2020-11) <a href="https://doi.org/gp574d">https://doi.org/gp574d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1142/9789811232701_0011">10.1142/9789811232701_0011</a></div></div>
</div>
<div id="ref-ynODVhC7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>How COVID-19 Is Changing Our Language : Detecting Semantic Shift in Twitter Word Embeddings</strong> <div class="csl-block">Yanzhu Guo, Christos Xypolopoulos, Michalis Vazirgiannis</div> <em>arXiv</em> (2021-02-17) <a href="https://arxiv.org/abs/2102.07836">https://arxiv.org/abs/2102.07836</a></div>
</div>
<div id="ref-K1SZY72U" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Natural Language Processing Reveals Vulnerable Mental Health Support Groups and Heightened Health Anxiety on Reddit During COVID-19: Observational Study</strong> <div class="csl-block">Daniel M Low, Laurie Rumker, Tanya Talkar, John Torous, Guillermo Cecchi, Satrajit S Ghosh</div> <em>Journal of Medical Internet Research</em> (2020-10-12) <a href="https://doi.org/ghm9v2">https://doi.org/ghm9v2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2196/22635">10.2196/22635</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32936777">32936777</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7575341">PMC7575341</a></div></div>
</div>
<div id="ref-18ZyyTcTe" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>PubTator central: automated concept annotation for biomedical full text articles</strong> <div class="csl-block">Chih-Hsuan Wei, Alexis Allot, Robert Leaman, Zhiyong Lu</div> <em>Nucleic Acids Research</em> (2019-05-22) <a href="https://doi.org/ggzfsc">https://doi.org/ggzfsc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkz389">10.1093/nar/gkz389</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31114887">31114887</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6602571">PMC6602571</a></div></div>
</div>
<div id="ref-11YUuHulp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>TaggerOne: joint named entity recognition and normalization with semi-Markov Models</strong> <div class="csl-block">Robert Leaman, Zhiyong Lu</div> <em>Bioinformatics</em> (2016-06-09) <a href="https://doi.org/f855dg">https://doi.org/f855dg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btw343">10.1093/bioinformatics/btw343</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27283952">27283952</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5018376">PMC5018376</a></div></div>
</div>
<div id="ref-aOPX10e0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>GNormPlus: An Integrative Approach for Tagging Genes, Gene Families, and Protein Domains</strong> <div class="csl-block">Chih-Hsuan Wei, Hung-Yu Kao, Zhiyong Lu</div> <em>BioMed Research International</em> (2015) <a href="https://doi.org/gb85jb">https://doi.org/gb85jb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1155/2015/918710">10.1155/2015/918710</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26380306">26380306</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4561873">PMC4561873</a></div></div>
</div>
<div id="ref-NXIFrudx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>SR4GN: A Species Recognition Software Tool for Gene Normalization</strong> <div class="csl-block">Chih-Hsuan Wei, Hung-Yu Kao, Zhiyong Lu</div> <em>PLoS ONE</em> (2012-06-05) <a href="https://doi.org/gpq498">https://doi.org/gpq498</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pone.0038460">10.1371/journal.pone.0038460</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22679507">22679507</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3367953">PMC3367953</a></div></div>
</div>
<div id="ref-17LQKv7vO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>tmVar 2.0: integrating genomic variant information from literature with dbSNP and ClinVar for precision medicine</strong> <div class="csl-block">Chih-Hsuan Wei, Lon Phan, Juliana Feltz, Rama Maiti, Tim Hefferon, Zhiyong Lu</div> <em>Bioinformatics</em> (2017-09-01) <a href="https://doi.org/gbzsmc">https://doi.org/gbzsmc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btx541">10.1093/bioinformatics/btx541</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28968638">28968638</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5860583">PMC5860583</a></div></div>
</div>
<div id="ref-q2fFAZTG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>&lt;span class="nocase"&gt;spaCy 2&lt;/span&gt;: Natural language understanding with Bloom embeddings, convolutional neural networks and incremental parsing</strong> <div class="csl-block">Matthew Honnibal, Ines Montani</div> (2017)</div>
</div>
<div id="ref-9gVmEi52" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>Machine access and text/data mining resources | bioRxiv</strong> <a href="https://www.biorxiv.org/tdm">https://www.biorxiv.org/tdm</a></div>
</div>
<div id="ref-v3H8juk7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>Machine access and text/data mining resources | medRxiv</strong> <a href="https://www.medrxiv.org/tdm">https://www.medrxiv.org/tdm</a></div>
</div>
<div id="ref-1GhHIDxuW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Efficient Estimation of Word Representations in Vector Space</strong> <div class="csl-block">Tomas Mikolov, Kai Chen, Greg Corrado, Jeffrey Dean</div> <em>arXiv</em> (2013-09-10) <a href="https://arxiv.org/abs/1301.3781">https://arxiv.org/abs/1301.3781</a></div>
</div>
<div id="ref-18rNo4lJI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>Factors Influencing the Surprising Instability of Word Embeddings</strong> <div class="csl-block">Laura Wendlandt, Jonathan K Kummerfeld, Rada Mihalcea</div> <em>arXiv</em> (2020-06-05) <a href="https://arxiv.org/abs/1804.09692">https://arxiv.org/abs/1804.09692</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18653/v1/N18-1190">10.18653/v1/n18-1190</a></div></div>
</div>
<div id="ref-8G9bCV2k" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>Stability of Word Embeddings Using Word2Vec</strong> <div class="csl-block">Mansi Chugh, Peter A Whigham, Grant Dick</div> <em>AI 2018: Advances in Artificial Intelligence</em> (2018) <a href="https://doi.org/gpxkkc">https://doi.org/gpxkkc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-3-030-03991-2_73">10.1007/978-3-030-03991-2_73</a></div></div>
</div>
<div id="ref-ImhJ75vq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>Evaluating the Stability of Embedding-based Word Similarities</strong> <div class="csl-block">Maria Antoniak, David Mimno</div> <em>Transactions of the Association for Computational Linguistics</em> (2018-12) <a href="https://doi.org/gf39k8">https://doi.org/gf39k8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1162/tacl_a_00008">10.1162/tacl_a_00008</a></div></div>
</div>
<div id="ref-bXXIQj3W" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>Predicting Word Embeddings Variability</strong> <div class="csl-block">Benedicte Pierrejean, Ludovic Tanguy</div> <em>Proceedings of the Seventh Joint Conference on Lexical and
Computational Semantics</em> (2018) <a href="https://doi.org/gh6qpc">https://doi.org/gh6qpc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18653/v1/s18-2019">10.18653/v1/s18-2019</a></div></div>
</div>
<div id="ref-NjbU71da" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>A generalized solution of the orthogonal procrustes problem</strong> <div class="csl-block">Peter H Schönemann</div> <em>Psychometrika</em> (1966-03) <a href="https://doi.org/dx77sz">https://doi.org/dx77sz</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/bf02289451">10.1007/bf02289451</a></div></div>
</div>
<div id="ref-157h5hA34" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction</strong> <div class="csl-block">Leland McInnes, John Healy, James Melville</div> <em>arXiv</em> (2020-09-21) <a href="https://arxiv.org/abs/1802.03426">https://arxiv.org/abs/1802.03426</a></div>
</div>
<div id="ref-TB6Pquc2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>Statistically Significant Detection of Linguistic Change</strong> <div class="csl-block">Vivek Kulkarni, Rami Al-Rfou, Bryan Perozzi, Steven Skiena</div> <em>Proceedings of the 24th International Conference on World Wide Web</em> (2015-05-18) <a href="https://doi.org/ghcv6k">https://doi.org/ghcv6k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1145/2736277.2741627">10.1145/2736277.2741627</a></div></div>
</div>
<div id="ref-1BaA3rWCK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>Improving semantic change analysis by combining word embeddings and word frequencies</strong> <div class="csl-block">Adrian Englhardt, Jens Willkomm, Martin Schäler, Klemens Böhm</div> <em>International Journal on Digital Libraries</em> (2019-05-20) <a href="https://doi.org/gpxkkd">https://doi.org/gpxkkd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s00799-019-00271-6">10.1007/s00799-019-00271-6</a></div></div>
</div>
<div id="ref-xQaNPZgV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>Diachronic Word Embeddings Reveal Statistical Laws of Semantic Change</strong> <div class="csl-block">William L Hamilton, Jure Leskovec, Dan Jurafsky</div> <em>arXiv</em> (2016) <a href="https://doi.org/gp8zpp">https://doi.org/gp8zpp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.48550/arxiv.1605.09096">10.48550/arxiv.1605.09096</a></div></div>
</div>
<div id="ref-1346YOPVE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>DUKweb, diachronic word representations from the UK Web Archive corpus</strong> <div class="csl-block">Adam Tsakalidis, Pierpaolo Basile, Marya Bazzi, Mihai Cucuringu, Barbara McGillivray</div> <em>Scientific Data</em> (2021-10-15) <a href="https://doi.org/gqbkx4">https://doi.org/gqbkx4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41597-021-01047-x">10.1038/s41597-021-01047-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34654827">34654827</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8520005">PMC8520005</a></div></div>
</div>
<div id="ref-YuxGez24" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>SARS: clinical virology and pathogenesis</strong> <div class="csl-block">John NICHOLLS, Xiao-Ping DONG, Gu JIANG, Malik PEIRIS</div> <em>Respirology</em> (2003-11) <a href="https://doi.org/cxjwrc">https://doi.org/cxjwrc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1046/j.1440-1843.2003.00517.x">10.1046/j.1440-1843.2003.00517.x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/15018126">15018126</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7169081">PMC7169081</a></div></div>
</div>
<div id="ref-1ExALQ5eN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) and coronavirus disease-2019 (COVID-19): The epidemic and the challenges</strong> <div class="csl-block">Chih-Cheng Lai, Tzu-Ping Shih, Wen-Chien Ko, Hung-Jen Tang, Po-Ren Hsueh</div> <em>International Journal of Antimicrobial Agents</em> (2020-03) <a href="https://doi.org/ggpj9d">https://doi.org/ggpj9d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.ijantimicag.2020.105924">10.1016/j.ijantimicag.2020.105924</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32081636">32081636</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7127800">PMC7127800</a></div></div>
</div>
<div id="ref-vvMItrwz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>The species Severe acute respiratory syndrome-related coronavirus: classifying 2019-nCoV and naming it SARS-CoV-2</strong> <em>Nature Microbiology</em> (2020-03-02) <a href="https://doi.org/ggqj7m">https://doi.org/ggqj7m</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41564-020-0695-z">10.1038/s41564-020-0695-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32123347">32123347</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7095448">PMC7095448</a></div></div>
</div>
<div id="ref-M1LzK6yf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>Factors Influencing the Surprising Instability of Word Embeddings</strong> <div class="csl-block">Laura Wendlandt, Jonathan K Kummerfeld, Rada Mihalcea</div> <em>arXiv</em> (2018) <a href="https://doi.org/gqcn9m">https://doi.org/gqcn9m</a> <div class="csl-block">DOI: <a href="https://doi.org/10.48550/arxiv.1804.09692">10.48550/arxiv.1804.09692</a></div></div>
</div>
<div id="ref-ZTyQYlDz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>arXiv.org: the Los Alamos National Laboratory e‐print server</strong> <div class="csl-block">Gerry McKiernan</div> <em>International Journal on Grey Literature</em> (2000-09-01) <a href="https://doi.org/fg8pw7">https://doi.org/fg8pw7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1108/14666180010345564">10.1108/14666180010345564</a></div></div>
</div>
<div id="ref-1D4Cz1GNr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline"><strong>PsyArXiv</strong> <div class="csl-block">Margie Ruppel</div> <em>The Charleston Advisor</em> (2021-10-01) <a href="https://doi.org/gqbk4k">https://doi.org/gqbk4k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.5260/chara.23.2.38">10.5260/chara.23.2.38</a></div></div>
</div>
</div>
<h2 id="supplemental-tables">Supplemental Tables</h2>
<div id="tbl:published_preprint_change_table" class="tablenos">
<table id="tbl:published_preprint_change_table" data-tag="S1">
<caption><span>Table S1:</span> The intersection of changepoints found between published papers and preprints. </caption>
<thead>
<tr class="header">
<th>Token</th>
<th>Changepoint</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>lockdown</td>
<td>2019-2020</td>
</tr>
<tr class="even">
<td>2021</td>
<td>2020-2021</td>
</tr>
<tr class="odd">
<td>distancing</td>
<td>2019-2020</td>
</tr>
<tr class="even">
<td>2019</td>
<td>2018-2019</td>
</tr>
<tr class="odd">
<td>ace2</td>
<td>2019-2020</td>
</tr>
<tr class="even">
<td>pandemic</td>
<td>2019-2020</td>
</tr>
<tr class="odd">
<td>2020</td>
<td>2019-2020</td>
</tr>
<tr class="even">
<td>coronavirus</td>
<td>2019-2020</td>
</tr>
<tr class="odd">
<td>bcl2a1</td>
<td>2018-2019</td>
</tr>
<tr class="even">
<td>peak3</td>
<td>2020-2021</td>
</tr>
<tr class="odd">
<td>3.6.2</td>
<td>2019-2020</td>
</tr>
<tr class="even">
<td>quarantine</td>
<td>2019-2020</td>
</tr>
<tr class="odd">
<td>cobl</td>
<td>2020-2021</td>
</tr>
<tr class="even">
<td>injectrode</td>
<td>2020-2021</td>
</tr>
<tr class="odd">
<td>nrc3</td>
<td>2020-2021</td>
</tr>
<tr class="even">
<td>4.0.5</td>
<td>2020-2021</td>
</tr>
<tr class="odd">
<td>TMPRSS2 (gene_7113)</td>
<td>2019-2020</td>
</tr>
<tr class="even">
<td>n262</td>
<td>2019-2020</td>
</tr>
<tr class="odd">
<td>bin1</td>
<td>2017-2018</td>
</tr>
<tr class="even">
<td>n3c</td>
<td>2020-2021</td>
</tr>
<tr class="odd">
<td>tip1</td>
<td>2020-2021</td>
</tr>
<tr class="even">
<td>omicron</td>
<td>2020-2021</td>
</tr>
<tr class="odd">
<td>pangolin</td>
<td>2019-2020</td>
</tr>
<tr class="even">
<td>adrn</td>
<td>2020-2021</td>
</tr>
<tr class="odd">
<td>seir</td>
<td>2019-2020</td>
</tr>
</tbody>
</table>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
